### PR TITLE
fix(onedrive): check response content type before trying to parse JSON

### LINKED ISSUE #151: Deleting a table row reports an incomplete JSON response
**Description**
If I execute `Table.deleteRow` I get a StatusCodeError with status 500, and a message `Unexpected end of JSON input`, although the call succeeded. The underlying issue is that `OneDrive.doFetch` expects the response to be valid JSON **even** if the response is empty (as is the case for this API: https://docs.microsoft.com/en-us/graph/api/tablerow-delete?view=graph-rest-1.0&tabs=http)

**To Reproduce**
1. Go to coralogix
2. Search for `Excel.delete`
3. Look at the warnings generated

**Expected behavior**
`Table.deleteRow` should not throw.

**Version:**
4.0.4

**Additional context:**
Has been introduced with the switch to helix-fetch (see #142)

### PR DESCRIPTION
**Pull request by dominique-pfister**
fix #151

### CODE REVIEW

File: src/OneDrive.js [hunk 1]
```diff
@@ -285,8 +285,10 @@ class OneDrive extends EventEmitter {
         }
         throw err;
       }
+      const contentType = resp.headers.get('content-type');
+      const json = contentType && contentType.startsWith('application/json');
       // await result in order to be able to catch any error
-      return await (rawResponseBody ? resp.buffer() : resp.json());
+      return await (rawResponseBody || !json ? resp.buffer() : resp.json());
```
Comment (tripodsan) [inline src/OneDrive.js:291]: in this case, the caller always needs to check if the response is a buffer or a (json) object.

**Review by tripodsan — CHANGES_REQUESTED** [review]
not sure about the new version. in any case, also adjust the documentation

**Review by stefan-guggisberg — APPROVED** [review]

**Comment by dominique-pfister** [comment]
> not sure about the new version. in any case, also adjust the documentation

I can also change the `Table.delete` to pass **true** for the `rawResponseBody` parameter, which would fix the issue as well.

**Comment by dominique-pfister** [comment]
> ... also adjust the documentation

OneDrive.doFetch is currently undocumented...

**Comment by dominique-pfister** [comment]
Changed code on the "client" side: identified all APIs that return an empty response, and therefore need no JSON parsing.

**Comment by tripodsan** [comment]
> > ... also adjust the documentation
> 
> `OneDrive.doFetch` is currently undocumented...

well then!

**Review by tripodsan — CHANGES_REQUESTED** [review]
well, I think throwing an error if the response is not a json is wrong.
then rather having the `buffer|object` ambiguety.

**Review by stefan-guggisberg — APPROVED** [review]

**Comment by stefan-guggisberg** [comment]
> > > ... also adjust the documentation
> > 
> > 
> > `OneDrive.doFetch` is currently undocumented...
> 
> well then!

@tripodsan The exposed `OneDrive` API always was and still is largely undocumented, see [here](https://github.com/adobe/helix-onedrive-support/blob/main/docs/API.md#OneDrive). This PR addresses a specific issue (#151). Lack of documentation should IMO be addressed in a separate issue/PR.

**Comment by dominique-pfister** [comment]
> well, I think throwing an error if the response is not a json is wrong.
> then rather having the `buffer|object` ambiguety.

fair enough, so I added that check back

**Comment by tripodsan** [comment]
> @tripodsan The exposed `OneDrive` API always was and still is largely undocumented, see [here](https://github.com/adobe/helix-onedrive-support/blob/main/docs/API.md#OneDrive). This PR addresses a specific issue (#151). Lack of documentation should IMO be addressed in a separate issue/PR.

I know... but when changing the behaviour of the api, we could gradually add more documentation.... at least adjust the signature in the types.

**Review by tripodsan — APPROVED** [review]

**Comment by stefan-guggisberg** [comment]
> at least adjust the signature in the types.

I agree, and that's what I did :-) 

https://github.com/adobe/helix-onedrive-support/commit/2c400724bd5e65a8953c0258d4f676e6b566d145#diff-e8aa1fc0f32bbe7c1f688d6f8259634a0fc7b046ec10b6ab66102847d51b632fR123-R125

**Pull request merged**